### DISCUSSION TITLE
Choose the number of NPCs to appear / disappear by the event

### DISCUSSION THREAD

**Discussion by Alexoune001** [AscensionGameDev/Intersect-Engine#1500] [Ideas]
**Is your feature request related to a problem? Please describe.**
No, it's for an idea of ​​improvement.

**Describe the solution you'd like**
In the event editor on the map, more precisely in the option to appear / disappear an NPC, to be able to directly choose the number of NPCs that you want to appear / disappear, then you will avoid copying / pasting as many times as 'we want the NPC to appear on the map

**Additional context**
Its particular avoid having a huge event page in the event that a person wants to make a summoning system in his game with a choice of NPCs (monster) for example

I don't know if you saw more or less what I'm asking.

**Comment by Khaikaa** [comment]
I'm not sure if there is a wasy way to do what you mean, where should all the npcs spawn? on the same spot? choosing a spot for each of them? in that case you already can just copy/paste that command. And what about despawning them? which ones should despawn? I don't think this could be done the way you ask.

What could be an actual improvement(but idk if it would be easy to implement) is the possibility of the despawn npc command only applying to the npcs spawn by the player who spawned them(the player who ran the event whish spawned them). That way we could add cool things such as summoning.

**Comment by Alexoune001** [comment]
Hmm I see, in this case make sure that the NPCs in question appear in the map but randomly?

For those who have to disappear, I suppose it will be in relation to the NPCs in question when we kill him, the one we killed disappears?

**Comment by WeylonSantana** [comment]
Only the spawn command should be chosen an amount of spawn, I've been through the situation of needing to spawn some monsters and have to copy and paste, a lot, which is very annoying.

About where to spawn, asked by khaikaa, there should be a configurable range, so that they spawn randomly within the range, if the range was 1, they would all spawn on the same tile.

About despawn, asked by khaikaa, the despawn npc command (as far as I can remember) kills all player generated npcs by default, and it should stay that way

that being said I totally agree
